aws/src/integration/java/org/apache/iceberg/aws/glue/DynamoLockManagerTest.java

```diff
@@ -0,0 +1,199 @@
+          .tableName(lockTableName)
+          .build());
+    } catch (Exception e) {
+      LOG.error("encountered failure", e);
+      Assert.fail("Should not throw exception when describing DynamoDB lock table");
+    }
+  }
+
+  @Test
+  public void testLock_singleProcess() {
+    boolean succeed = lockManager.tryLock(tableId(tableName),
+        AwsProperties.LOCK_EXPIRE_MS_DEFAULT);
+    Assert.assertTrue(succeed);
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put("lockId", AttributeValue.builder().s(tableId(tableName)).build());
+    try {
+      dynamo.getItem(GetItemRequest.builder()
+          .tableName(lockTableName)
+          .key(key)
+          .build());
+    } catch (Exception e) {
+      LOG.error("encountered failure", e);
+      Assert.fail("Should not throw exception when getting lock after insert");
+    }
+  }
+
+  @Test
+  public void testLock_multiProcess() {
+    List<Boolean> results = IntStream.range(0, 100).parallel()
+        .mapToObj(i -> lockManager.tryLock(tableId(tableName),
+            AwsProperties.LOCK_EXPIRE_MS_DEFAULT))
+        .collect(Collectors.toList());
+    Assert.assertEquals(1, results.stream().filter(s -> s).count());
+  }
+
+  @Test
+  public void testWaitLock_singleProcess() throws Exception {
+    boolean succeed = lockManager.tryLock(tableId(tableName),
+        AwsProperties.LOCK_EXPIRE_MS_DEFAULT,
+        AwsProperties.LOCK_WAIT_MS_DEFAULT);
+    Assert.assertTrue(succeed);
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put("lockId", AttributeValue.builder().s(tableId(tableName)).build());
+    try {
+      dynamo.getItem(GetItemRequest.builder()
+          .tableName(lockTableName)
+          .key(key)
+          .build());
+    } catch (Exception e) {
+      LOG.error("encountered failure", e);
+      Assert.fail("Should not throw exception when getting lock after insert");
+    }
+  }
+
+  @Test
+  public void testWaitLock_multiProcess_allSucceed() {
+    List<Boolean> results = IntStream.range(0, 100).parallel()
+        .mapToObj(i -> {
+          try {
+            return lockManager.tryLock(tableId(tableName), 10, 100000);
+          } catch (InterruptedException e) {
+            LOG.error("lock acquire interrupted", e);
+            return false;
+          }
+        })
+        .collect(Collectors.toList());
+    Assert.assertEquals(100, results.stream().filter(s -> s).count());
+  }
+
+  @Test
+  public void testWaitLock_multiProcess_allFail() {
+    List<Boolean> results = IntStream.range(0, 100).parallel()
+        .mapToObj(i -> {
+          try {
+            return lockManager.tryLock(tableId(tableName), 1000000, 10);
+          } catch (InterruptedException e) {
+            LOG.error("lock acquire interrupted", e);
+            return false;
+          }
+        })
+        .collect(Collectors.toList());
+    Assert.assertEquals(1, results.stream().filter(s -> s).count());
+  }
+
+
+  @Test
+  public void testUnlock() {
```

Comment:
This doesn't test unlock, it tests that lock succeeds after dropping the lock item in the table.